packages/main/src/plugin/container-registry.spec.ts

```diff
@@ -1220,6 +1220,39 @@ test('pull unknown image fails with error 403', async () => {
   );
 });
 
+test('pulling an image with platform=linux/arm64 will add the platform to the pull options', async () => {
+  const getMatchingEngineFromConnectionSpy = vi.spyOn(containerRegistry, 'getMatchingEngineFromConnection');
+  //vi.spyOn(containerRegistry, 'pullImage').mockReturnValue(Promise.resolve());
+
+  const pullMock = vi.fn();
+
+  const fakeDockerode = {
+    pull: pullMock,
+    modem: {
+      followProgress: vi.fn(),
+    },
+  } as unknown as Dockerode;
+
+  getMatchingEngineFromConnectionSpy.mockReturnValue(fakeDockerode);
+
+  const containerConnectionInfo = {} as ProviderContainerConnectionInfo;
+
+  pullMock.mockResolvedValue({ statusCode: 201 });
+
+  const callback = vi.fn();
+
+  // Expect pull image to pass
+  // we use .catch to avoid an unhandled promise rejection, we don't care about the result, we just
+  // want to check that the call was made with the correct platform
+  containerRegistry.pullImage(containerConnectionInfo, 'unknown-image', callback, 'linux/arm64').catch(() => {});
```

Comment:
it's still odd to me why the mocking is failing ? It shouldn't throw any exception

Just calling await and that's all

Reply:
Can't :( we time out no matter what mocking I do, so suggested we just do a `catch` with it.